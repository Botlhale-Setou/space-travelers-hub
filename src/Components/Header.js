import React from 'react';
import Nav from './Nav';
import logo from '../assets/logo.png';

function Header() {
  return (
    <div className="header">
      <div className="headerMain" data-testid="header">
        <img src={logo} alt="Logo" />
        <h1 data-testid="siteTitle">Space Travelers&apos; Hub</h1>
      </div>
      <Nav />
    </div>
  );
}

export default Header;
