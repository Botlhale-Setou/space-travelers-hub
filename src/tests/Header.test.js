import React from 'react';
import Renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import Header from '../Components/Header';

describe('Header', () => {
  test('renders correctly', () => {
    const tree = Renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the correct header', () => {
    const { getByTestId } = render(
      <Router>
        <Header />
      </Router>,
    );
    const header = getByTestId('header');
    expect(header).toBeInTheDocument();
  });

  test('renders the correct title', () => {
    const { getByTestId } = render(
      <Router>
        <Header />
      </Router>,
    );
    const siteTitle = getByTestId('siteTitle');
    expect(siteTitle).toBeInTheDocument();
  });
});
