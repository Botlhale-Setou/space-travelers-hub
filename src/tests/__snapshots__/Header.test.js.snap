// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="header"
>
  <div
    className="headerMain"
    data-testid="header"
  >
    <img
      alt="Logo"
      src="logo.png"
    />
    <h1
      data-testid="siteTitle"
    >
      Space Travelers' Hub
    </h1>
  </div>
  <ul
    className="nav"
  >
    <li>
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className="link"
        href="/Missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        className="link"
        href="/MyProfile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</div>
`;
